### PR TITLE
Reduce docker image size on build

### PR DESCRIPTION
**Pull request by stdevMac**
## Description

Currently, the docker image has a size of almost `580MB`

```
juno                                new        3c4310b2103a   5 minutes ago   371MB
juno                                latest          ad813212fccf   4 days ago      586MB
```

## Changes:

- Reduce python dependencies on the image
- Add more stages to build making fewer layers

## Types of changes

_Leave only items that describe your changes, remove the rest. Remove this line too._

- Refactoring (no functional changes, no api changes)
- Build related changes

### CODE REVIEW

**Review by adriantpaez — APPROVED** [review]
I built the image and it works 👍

File: Dockerfile [hunk 1]
```diff
@@ -12,20 +13,37 @@ RUN go mod download
 
 RUN make juno
 
-FROM python:3.7.13-alpine
+# State 2: Build Python dependencies for Cairo
```
Comment (tshakalekholoane) [inline Dockerfile:16]: Typo.

```suggestion
# Stage 2: Build Python dependencies for Cairo
```

File: Dockerfile [hunk 2]
```diff
@@ -12,20 +13,37 @@ RUN go mod download
...
 
 WORKDIR /app
 
-COPY --from=build /app/build/juno /home/app/juno
-COPY --from=build /app/requirements.txt /req/requirements.txt
+COPY ./requirements.txt /req/requirements.txt
 
-RUN apk update && apk upgrade && apk add --update alpine-sdk && \
-    apk add --no-cache gmp-dev cmake gcc g++ linux-headers && pip install -r /req/requirements.txt
+# Install Python Dependencies
```
Comment (tshakalekholoane) [inline Dockerfile:23]: Use consistent formatting.

```suggestion
# Install Python dependencies
```

File: Dockerfile [hunk 3]
```diff
@@ -12,20 +13,37 @@ RUN go mod download
...
+RUN apk update && apk upgrade && apk add --update alpine-sdk && apk add --no-cache gmp-dev cmake gcc g++ linux-headers
+
+# Install project Dependencies
```
Comment (tshakalekholoane) [inline Dockerfile:26]: Use consistent formatting.

```suggestion
# Install project dependencies
```

File: Dockerfile [hunk 4]
```diff
@@ -12,20 +13,37 @@ RUN go mod download
...
+RUN pip --disable-pip-version-check install -r /req/requirements.txt
+ENV PY_PATH=/usr/local/lib/python3.7/
+RUN find ${PY_PATH} -type d -a -name test -exec rm -rf '{}' + \
+    && find ${PY_PATH} -type d -a -name tests  -exec rm -rf '{}' + \
+    && find ${PY_PATH} -type f -a -name '*.pyc' -exec rm -rf '{}' + \
+    && find ${PY_PATH} -type f -a -name '*.pyo' -exec rm -rf '{}' +
+
+# Stage 3: Build Docker Image
```
Comment (tshakalekholoane) [inline Dockerfile:34]: Use consistent formatting.

```suggestion
# Stage 3: Build Docker image
```

**Review by tshakalekholoane — APPROVED** [review]
LGTM. See suggestions for minor cosmetic changes.

**Review by joshklop — APPROVED** [review]

**Pull request merged**